<a href="https://colab.research.google.com/github/GezhinOleg/Py_Stat_L3/blob/main/random_var.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Смоделировать игру против лотерейного автомата типа “777”. Игрок платит 1 руб., после чего выпадает случайное целое число, равномерно распределенное от 0 до 999. При некоторых значениях числа игрок получает выигрыш.

    777: 200 руб.
    999: 100 руб.
    555: 50 руб.
    333: 15 руб.
    111: 10 руб.
    *77: 5 руб.
    **7: 3 руб.
    *00: 2 руб.
    **0: 1 руб.

Вопросы:

    Выгодна ли игра игроку?
    Сколько в среднем приобретает или теряет игрок за одну игру?

**Дополнительное задание повышенной сложности. **

Теоретически рассчитать средний выигрыш (проигрыш) и сравнить с результатами моделирования

    – любая цифра


In [1]:
import numpy as np
trials = 1000000    # Количество попыток
successes = 0    # Количество успехов
finance_start = 1000000 # сумма которой игрок зашел в казино

for i in range(trials):
    finance_start -= 1
    random_score = np.random.randint(0, 999)    # Случайное количество очков от 0 до 999
    if random_score == 777:
        successes += 1
        finance_start += 200
        
    elif random_score != 777 and random_score % 100 == 77:
        successes += 1
        finance_start += 5
        
    elif random_score != 777 and random_score % 100 != 77 and random_score % 10 == 7:
        successes += 1
        finance_start += 3
        
    elif random_score == 999:
        successes += 1
        finance_start += 100
        
    elif random_score == 555:
        successes += 1
        finance_start += 50
        
    elif random_score == 333:
        successes += 1
        finance_start += 30
        
    elif random_score == 111:
        successes += 1
        finance_start += 15
        
    elif random_score % 10 == 0:
        successes += 1
        finance_start += 1
        
    elif random_score % 100 == 0:
        successes += 1
        finance_start += 2
        
print('На начало игры было: 1 000 000 рублей')        
print('Количество успехов:', successes)
print('Остаток денег после 1000000 ставок:', finance_start)
print('Процент выпадения выигрышных комбинаций:', successes / trials*100, '%')
print('Получено за игру:', (finance_start - trials) / trials)

На начало игры было: 1 000 000 рублей
Количество успехов: 202754
Остаток денег после 1000000 ставок: 699613
Процент выпадения выигрышных комбинаций: 20.275399999999998 %
Получено за игру: -0.300387


In [ ]:
count = 0

for i in range(1000):
    if i == 777 or i == 999 or i == 555 or i == 333  or i == 111 or i%10 == 0 or i%100 == 0:
        count += 1
    elif i != 777 and i%100 == 77:
        count += 1
    elif i != 777 and i%100 != 77 and i%10 == 7:
        count += 1
print('Вероятность выпадения выигрышных комбинаций', count/1000*100, '%')

Вероятность выпадения выигрышных комбинаций 20.4 %


Вероятность выпадения выигрышной комбинации - 20.4 %.
Согласно практики выигрышные комбинации из 1 миллиона повторений выпадают приблизительно с такой же частотой. С учетом ставки на каждую игру 1 рубль, в итоге игрок за каждую игру теряет приблизительно 30 копеек. 
Азартные игры ЗЛО!!!